aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/CreateHandler.java

```diff
@@ -60,6 +66,8 @@
                 ResourceModel.TYPE_NAME, pgName, awsAccountId));
         }
 
+        addChannelNotifications(pgName, channelUris, proxy, profilerClient);
```

Comment:
This call should work in a similar way to `putAgentPermissions` below: If the channel addition fails, it should delete the profiling group, so that customers don't get stuck in a weird state where they manually need to go in the console and delete the profiling group.

The behavior we're going for here is that this tries as much as possible (given the current API) to appear atomic: either the whole create succeeds, permissions and notifications, or none of it does and we delete all of it.